cypress/e2e/checkout.cy.js

```diff
@@ -1,11 +1,47 @@
+import CartPageObject from '../support/pages/cartPage.pageObject';
+import HomeAndCataloguePageObject
+  from '../support/pages/homeСatalogue.pageObject';
+
 /// <reference types='cypress' />
 
-describe('', () => {
-  before(() => {
+const homePage = new HomeAndCataloguePageObject();
+const cartPage = new CartPageObject();
 
-  });
+const testData = {
+  category: 'Laptops',
+  product: 'Sony vaio i7',
+  alertMessage: 'Product added',
+  link: 'Cart',
+  successMessage: 'Thank you for your purchase!'
+};
 
-  it('', () => {
+describe('Demoblaze', () => {
+  let user;
+  before(() => {
+    cy.task('generateUser').then((generatedUser) => {
+      user = generatedUser;
+    });
+  });
 
+  it('should provide the ability to buy product', () => {
+    homePage.visit();
+    homePage.clickOnCategory(testData.category);
+    homePage.clickOnProduct(testData.product);
+    homePage.clickOnCart();
+    homePage.assertAllert(testData.alertMessage);
+    homePage.clickOnLink(testData.link);
+    cartPage.productAssert.should('contain', testData.product);
```

Comment:
```suggestion
    cartPage.assertProduct(testData.product);
```
change this function into the page object method